### PR TITLE
Microverse final capstone frontend react codebase

### PR DESCRIPTION
**Pull request by samroodali**
Please review my final capstone. 

Requirements:
Create a React front-end with three pages
- [x] Items page -> All Teachers page
- [x] Single item  -> Teacher show page
- [x] Appointments -> Current user's appointment's page

Extra nice to have requirement:
- [x] Proper authentication with JWT Token stored in cookies
- [x] User's table with user's appointments only being shown

Technologies used:
* React
* React query
* Tailwind
* Framer motion

Thank you for your time and consideration.

### CODE REVIEW

File: src/api/rails.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
```
Comment (rvvergara) [inline src/api/rails.js:1]: Great job importing axios on only 1 file

File: src/api/rails.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+const rails = axios.create({
+  baseURL: 'http://localhost:5000/api',
```
Comment (rvvergara) [inline src/api/rails.js]: I would have used an environment file to store URL's. URL's may differ from environment to environment.

File: src/components/Signin.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate, Link } from 'react-router-dom';
+import useActions from '../hooks/useActions';
+
+const Signin = () => {
+  const [email, setEmail] = useState('samrood@gmailexample22.com');
+  const [password, setPassword] = useState('password');
```
Comment (rvvergara) [inline src/components/Signin.jsx]: What do you think about just setting an empty string as default state here?

File: src/components/Signin.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate, Link } from 'react-router-dom';
+import useActions from '../hooks/useActions';
+
+const Signin = () => {
+  const [email, setEmail] = useState('samrood@gmailexample22.com');
```
Comment (rvvergara) [inline src/components/Signin.jsx]: Probably an empty string is a better initial state, don't you think?

File: src/components/Signup.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import { useNavigate, Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import useActions from '../hooks/useActions';
+
+const Signup = () => {
+  const [username, setUsername] = useState('Samrood');
+  const [email, setEmail] = useState('samrood@gmailexample22.com');
```
Comment (rvvergara) [inline src/components/Signup.jsx]: Empty strings are better initial states for your username, email and password

File: src/components/horizontal-scroll/Card.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// import { VisibilityContext } from 'react-horizontal-scrolling-menu';
```
Comment (rvvergara) [inline src/components/horizontal-scroll/Card.jsx]: Let's remove commented out code

File: src/components/horizontal-scroll/Card.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  'green',
+  'purple',
+  'pink',
+  'blue',
+];
+
+export default function Card({
+  onClick,
+  item,
+  // itemId,
```
Comment (rvvergara) [inline src/components/horizontal-scroll/Card.jsx]: Remove commented out code

File: src/components/horizontal-scroll/Card.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+Card.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  item: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    index: PropTypes.number.isRequired,
+  }).isRequired,
+  // itemId: PropTypes.string.isRequired,
```
Comment (rvvergara) [inline src/components/horizontal-scroll/Card.jsx]: Remove commented out code

File: src/redux/reducers/current-user.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  SET_CURRENT_USER_ERROR,
+  SIGN_OUT,
+} from '../action-types';
+
+const initialState = {
+  username: '',
+  errorMessages: [],
+};
+
+const currentUser = (state = initialState, action) => {
```
Comment (rvvergara) [inline src/redux/reducers/current-user.js:12]: Great job in designing this reducer!

**Review by rvvergara — CHANGES_REQUESTED** [review]
Hi Sam, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Please note that this is still a 1.0 review and you should submit your code re-review request as a 1.0

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/api/rails.js (hunk 1 above)
Comment (samroodali) [reply src/api/rails.js:1]: Thank you @rvvergara

File: src/api/rails.js (hunk 2 above)
Comment (samroodali) [reply src/api/rails.js]: I agree. Thanks for the tip

File: src/components/Signup.jsx (hunk 5 above)
Comment (samroodali) [reply src/components/Signup.jsx]: Thanks. Good catch. I have removed these default state.

File: src/components/horizontal-scroll/Card.jsx (hunk 7 above)
Comment (samroodali) [reply src/components/horizontal-scroll/Card.jsx]: I have removed them :+1:

File: src/components/horizontal-scroll/Card.jsx (hunk 8 above)
Comment (samroodali) [reply src/components/horizontal-scroll/Card.jsx]: Done :+1:

File: src/redux/reducers/current-user.js (hunk 9 above)
Comment (samroodali) [reply src/redux/reducers/current-user.js:12]: Thank you :smile:

File: src/api/rails.js (hunk 1 above)
Comment (rvvergara) [reply src/api/rails.js:1]: Welcome!

File: src/components/Signup.jsx (hunk 5 above)
Comment (rvvergara) [reply src/components/Signup.jsx]: Awesome!

File: src/components/horizontal-scroll/Card.jsx (hunk 7 above)
Comment (rvvergara) [reply src/components/horizontal-scroll/Card.jsx]: Cool!

**Comment by rvvergara** [comment]
Hi @SamroodAli , 

In my opinion, your front end is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by samroodali** [comment]
Thanks @rvvergara for your time and consideration. This has been a great learning experience.

**Pull request merged**